rama-http-backend/src/client/proxy/layer/proxy_connector/layer.rs

```diff
@@ -27,14 +32,21 @@ impl HttpProxyConnectorLayer {
     /// [`Context`]: rama_core::Context
     /// [`ProxyAddress`]: rama_net::address::ProxyAddress
     pub fn required() -> Self {
-        Self { required: true }
+        Self {
+            required: true,
+            version: None,
+        }
     }
 }
 
 impl<S> Layer<S> for HttpProxyConnectorLayer {
     type Service = HttpProxyConnector<S>;
 
     fn layer(&self, inner: S) -> Self::Service {
-        HttpProxyConnector::new(inner, self.required)
+        let mut svc = HttpProxyConnector::new(inner, self.required);
+        self.version.inspect(|version| {
```

Comment:
This logic will need to be different based on my other comment, as otherwise you'll miss the explicit `None == auto` choice.